internal/pipeline/pipeline.go

```diff
@@ -60,6 +73,69 @@ type Pipeline struct {
 	Validators []Function `yaml:"validators,omitempty"`
 }
 
+// String returns the string representation of Pipeline
+func (p *Pipeline) String() (string, error) {
+	b, err := yaml.Marshal(p)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to convert pipeline to string")
```

Comment:
Since Go 1.13 onwards, Go supports wrapping/unwrapping errors and also provides some helpers methods in errors package for ex. errors.Is(..), errors.As(...) that makes it easy to deal with errors. Pl. use that for wrapping errors here. Here is a blog post that talks about it https://blog.golang.org/go1.13-errors. Highly recommended, it also improves the errors handling code readability as well.